### PR TITLE
Add feature to specify obs data dir for ungridded reading and also in low and highlevel colocation

### PR DESCRIPTION
**Pull request by jgliss**
This also includes some minor cleanup that is unrelated.

### CODE REVIEW

**Comment by jgliss** [comment]
Missing tests for gridded gridded colocation

**Comment by jgliss** [comment]
And need to remove BLAAAAAAAAAAAAAAAAA prints in lowlevel colocation when functionality is confirmed.

**Comment by jgliss** [comment]
@ejgal can you check?

**Comment by jgliss** [comment]
> @ejgal can you check?

Is a bit more than initially planned ;)

**Comment by ejgal** [comment]
@jgliss, I have to run in a few minutes, but can have a look tomorrow morning.

**Comment by jgliss** [comment]
> @jgliss, I have to run in a few minutes, but can have a look tomorrow morning.

that's totally fine :)

File: pyaerocom/colocation_auto.py [hunk 1]
```diff
@@ -629,14 +652,47 @@ def _eval_resample_how(self, model_var, obs_var):
...
+        -------
+        None.
+
+        """
+        try:
+            yrs_avail = reader.years_avail
+        except AttributeError:
+            raise AttributeError('Input reader {} does not have attr. '
+                                 'years_avail') # not sure if ReadEMEP has...
+
```
Comment (ejgal) [inline pyaerocom/colocation_auto.py:679]: Yep, no years_avail yet in EMEP.

File: pyaerocom/test/test_colocation_auto.py [hunk 2]
```diff
@@ -124,12 +124,8 @@ def test_colocator_read_ungridded():
...
     col.read_opts_ungridded = {'last_file' : 10}
-    with pytest.raises(DataCoverageError):
-        data = col.read_ungridded(obs_var) # Why is this raised?
-        # read_ungridded expects a list of one or more variables!
-        # Should check for str and convert to list?
-        # Should read_ungridded and read_model_data have same arguments?
-    data = col.read_ungridded([obs_var])
+
+    data = col.read_ungridded(obs_var)
     assert isinstance(data, UngriddedData)
```
Comment (ejgal) [inline pyaerocom/test/test_colocation_auto.py:129]: Thanks for fixing this in col.read_ungridded :)

**Pull request merged**